google-cloud-bigquery/lib/google/cloud/bigquery/query_job.rb

```diff
@@ -106,6 +106,18 @@ def bytes_processed
         end
 
         ##
+        # Describes the execution plan for the query.
+        #
+        # @return [Array<Google::Cloud::Bigquery::QueryJob::Stage>]
+        #
+        def query_plan
+          return nil unless @gapi.statistics.query.query_plan
```

Comment:
The existing readers already assume these will always be there. And I think that's correct wrt the API. However, `query_plan` is only available for 7 days per the docs, and therefore may be `nil`.

```ruby
##
        # Checks if the query results are from the query cache.
        def cache_hit?
          @gapi.statistics.query.cache_hit
        end
```